packages/element-lib/src/__tests__/Production.Tests.spec.js

```diff
@@ -0,0 +1,110 @@
+const dotenv = require('dotenv');
+const path = require('path');
+
+const element = require('../../index');
+
+dotenv.config({ path: path.resolve(__dirname, '../../../../.env') });
+
+const config = {
+  mnemonic: process.env.ELEMENT_MNEMONIC,
+  web3ProviderUrl: process.env.ELEMENT_PROVIDER,
+  ipfsApiMultiAddr: process.env.ELEMENT_IPFS_MULTIADDR,
+  anchorContractAddress: process.env.ELEMENT_CONTRACT_ADDRESS,
+  couchdbRemote: process.env.ELEMENT_COUCHDB_REMOTE,
+};
+
+// These tests are for debuging syncing issues in production
+describe.skip('Production Tests', () => {
+  const blockchain = element.blockchain.ethereum.configure({
+    hdPath: "m/44'/60'/0'/0/0",
+    mnemonic: config.mnemonic,
+    providerUrl: config.web3ProviderUrl,
+    // when not defined, a new contract is created.
+    anchorContractAddress: config.anchorContractAddress,
+  });
+
+  const storage = element.storage.ipfs.configure({
+    multiaddr: config.ipfsApiMultiAddr,
+  });
+
+  const db = new element.adapters.database.ElementRXDBAdapterWithRemoteSync({
+    name: 'production-tests',
+    remote: config.couchdbRemote,
+  });
+
+  const manager = new element.adapters.storage.StorageManager(db, storage, {
+    autoPersist: true,
+    retryIntervalSeconds: 1,
+  });
+  describe('blockchain', () => {
+    it('should have 2 transactions in window', async () => {
+      await blockchain.resolving;
+      const txns = await blockchain.getTransactions(6421342, 6421354);
+      expect(txns.length).toBe(2);
+    });
+  });
+
+  describe('storage', () => {
+    it('anchor file is resolvable', async () => {
+      const anchorFile = await storage.read('QmPyAucuooAEMbdw1uT8veXrLSfBjdNoTqfDRsnt5hrzbS');
+      expect(anchorFile).toEqual({
+        batchFileHash: 'QmZxFCGwiYfutyXDR2TcWs728GrCeAsbVkM7Dazphf35QF',
+        didUniqueSuffixes: ['WJl6h_dz3bvJscpWceWoK7AbD5khKdob0W92zh1sx7M'],
+        merkleRoot: 'f0e895ea0791e90f3687498d16824299c9346dab93e6463c182b5b649d3e26ab',
+      });
+    });
+  });
+
+  describe('cloudant', () => {
+    it('can await sync', async () => {
+      const record = await db.read(
+        'element:sidetree:cas-cachable:QmPyAucuooAEMbdw1uT8veXrLSfBjdNoTqfDRsnt5hrzbS',
+      );
+      expect(record).toBe(null);
+      const record1 = await manager.read('QmPyAucuooAEMbdw1uT8veXrLSfBjdNoTqfDRsnt5hrzbS');
+      expect(record1).toEqual({
+        batchFileHash: 'QmZxFCGwiYfutyXDR2TcWs728GrCeAsbVkM7Dazphf35QF',
+        didUniqueSuffixes: ['WJl6h_dz3bvJscpWceWoK7AbD5khKdob0W92zh1sx7M'],
+        merkleRoot: 'f0e895ea0791e90f3687498d16824299c9346dab93e6463c182b5b649d3e26ab',
+      });
+      const record2 = await db.read(
+        'element:sidetree:cas-cachable:QmPyAucuooAEMbdw1uT8veXrLSfBjdNoTqfDRsnt5hrzbS',
+      );
+      expect(record2.persisted).toBe(true);
+      await manager.db.awaitableSync();
+    });
+  });
+
+  // TODO: Fix this code to test creation of a production DID End to End.
```

Comment:
We should track this in a ticket

Reply:
https://github.com/decentralized-identity/element/issues/90